manageability-and-operations/observability-and-manageability/external-database-enablement/root_module/modules/container/container_connection/variables.tf

```diff
@@ -0,0 +1,86 @@
+#Common database system variables
+
+variable "compartment_ocid" {
+  type = string
+}
+
+variable "external_database_connector_connection_string_hostname" {
+  description = "Hostname for connection string"
+  type        = string
+}
+
+variable "external_database_connector_connection_string_port" {
+  description = "Port for connection string"
+  type        = number
+  default     = 1521
+}
+
+variable "external_database_connector_connection_string_protocol" {
+  description = "Protocol for connection string. Can be 'TCP' or 'TCPS'"
+  type        = string
+  validation {
+    condition     = contains(["TCP", "TCPS"], var.external_database_connector_connection_string_protocol)
+    error_message = "The value of 'protocol' in the JSON-input for database systems must be 'TCP' or 'TCPS'"
+  }
+}
+
+variable "ssl_secret_id" {
+  description = "The OCID for the OCI Vault Secret used for TCPS-connections. PKCS12 and JKS are supported as trust store types"
+  type        = string
+  default     = null
+}
+
+variable "external_database_connector_agent_id" {
+  description = "The OCID for the management agent used for database connections"
+  type        = string
+  #default = "ocid1.managementagent.oc1..XXXXXXX"
+}
+
+#Container database variables
+
+variable "external_container_database_display_name" {
+  description = "Name for external container database"
+  type        = string
+}
+
+variable "external_cdb_connector_connection_credentials_credential_name" {
+  description = "Name of credential set for container database connector. Must be in a 'x.y' format"
+  type        = string
+  validation {
+    condition     = can(regex("[a-zA-Z\\d_]{1,64}\\.[a-zA-Z\\d_]{1,199}", var.external_cdb_connector_connection_credentials_credential_name))
+    error_message = "The value of var.external_cdb_connector_connection_credentials_credential_name must be in 'x.y' format. x has a maximum of 64 characters, and y has a maximum of 199 characters. The strings can only contain letters, numbers, and underscores"
+  }
+}
+
+variable "external_cdb_connector_connection_credentials_credential_type" {
+  description = "Set to 'DETAILS' for TCP credentials and 'SSL_DETAILS' for TCPS credentials"
+  type        = string
+  validation {
+    condition     = contains(["DETAILS", "SSL_DETAILS"], var.external_cdb_connector_connection_credentials_credential_type)
+    error_message = "The value of var.external_cdb_connector_connection_credentials_credential_type must be 'DETAILS' or 'SSL_DETAILS'"
+  }
+}
+
+variable "external_cdb_connector_connection_credentials_password" {
```

Comment:
How about adding a validation? Like value != "<INSERT PASSWORD HERE>"?

Reply:
Good idea. Have provided !="" validation for credential and connection variables to make sure it´s set to something before applying anything (See commit: "[Connection and credential variables validated for empty strings](https://github.com/oracle-devrel/technology-engineering/pull/856/commits/3e6cfab06d4288d3828b5b14db9d37ad82bc680d)"